### PR TITLE
feat: css-media

### PR DESCRIPTION
**Pull request by barak007**
refactor: use only one walk in transform
refactor: move valueHook to evaluator

### CODE REVIEW

File: packages/core/src/functions.ts [hunk 1]
```diff
@@ -40,21 +39,29 @@ export interface EvalValueResult {
...
+        getResolvedSymbols: (meta: StylableMeta) => MetaResolvedSymbols;
+    }) {
+        this.valueHook = options.valueHook;
         this.stVarOverride = options.stVarOverride;
+        this.getResolvedSymbols = options.getResolvedSymbols;
     }
     evaluateValue(
-        context: FeatureTransformContext,
-        data: Omit<EvalValueData, 'passedThrough'> & { passedThrough?: string[] }
+        context: Omit<FeatureTransformContext, 'getResolvedSymbols'>,
```
Comment (idoros) [inline packages/core/src/functions.ts:54]: you just added `passedThrough` to the transform context, and it also sounds like it belongs there more then to data.

Maybe pass is as part of the context and not the data?

File: packages/core/src/stylable-transformer.ts [hunk 2]
```diff
@@ -178,44 +183,39 @@ export class StylableTransformer {
...
 
-        ast.walkRules((rule) => {
-            if (isChildOfAtRule(rule, 'keyframes')) {
-                return;
-            }
-            rule.selector = this.scopeRule(meta, rule, topNestClassName);
-        });
-
-        ast.walkAtRules((atRule) => {
+        const handleAtRule = (atRule: postcss.AtRule) => {
```
Comment (idoros) [inline packages/core/src/stylable-transformer.ts:211]: would be nice to extract this functions

**Review by idoros — APPROVED** [review]
looks good

File: packages/core/src/stylable-transformer.ts (hunk 2 above)
Comment (barak007) [reply packages/core/src/stylable-transformer.ts:211]: I tried but it yields bigger change.

File: packages/core/src/functions.ts (hunk 1 above)
Comment (idoros) [reply packages/core/src/functions.ts:53]: I added this in the last commit, and added a related issue I fount that wan't caused by any change in this PR: https://github.com/wix/stylable/issues/2820

**Pull request merged**